### PR TITLE
[NFI]: Use namespace onnx_mlir rather than mlir

### PR DESCRIPTION
**Pull request by etiotto**
We decided (https://github.com/onnx/onnx-mlir/issues/1174) that we should not use the `mlir` namespace (or the global namespace) for our ONNX lowering passes, IndexExpr, exc. 

This PR remove the use of the mlir namespace and introduces the use of the `onnx_mlir` namespace instead. 
This PR does not make any functional changes.

### CODE REVIEW

File: src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp [hunk 1]
```diff
@@ -390,3 +394,5 @@ void populateLoweringONNXLoopOpPattern(RewritePatternSet &patterns,
     TypeConverter &typeConverter, MLIRContext *ctx) {
   patterns.insert<ONNXLoopOpLowering>(typeConverter, ctx);
 }
+
+} // namespace onnx_mlir
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp:398]: please fix.

File: src/Conversion/ONNXToKrnl/ConvertONNXToKrnl.cpp [hunk 2]
```diff
@@ -55,73 +57,99 @@ void populateONNXToKrnlConversionPattern(RewritePatternSet &patterns,
 
   // Frontend operation lowering.
   // ControlFlow
-  populateLoweringONNXLoopOpPattern(patterns, typeConverter, ctx);
-  populateLoweringONNXScanOpPattern(patterns, typeConverter, ctx);
+  onnx_mlir::populateLoweringONNXLoopOpPattern(patterns, typeConverter, ctx);
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/ConvertONNXToKrnl.cpp:60]: You already have `using namespace onnx_mlir;` at line 22, but do you need to prefix 1populateLoweringONNX.*Pattern` with `onnx_mlir::`?

File: src/Conversion/ONNXToKrnl/Math/MatMul.cpp [hunk 3]
```diff
@@ -325,3 +327,5 @@ void populateLoweringONNXMatMulOpPattern(RewritePatternSet &patterns,
     TypeConverter &typeConverter, MLIRContext *ctx, bool enableTiling) {
   patterns.insert<ONNXMatMulOpLowering>(typeConverter, ctx, enableTiling);
 }
+
+} // namespace onnx_mlir
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/Math/MatMul.cpp:331]: Please fix.

File: src/Conversion/ONNXToKrnl/Math/Softmax.cpp [hunk 4]
```diff
@@ -286,3 +288,5 @@ void populateLoweringONNXSoftmaxOpPattern(RewritePatternSet &patterns,
     TypeConverter &typeConverter, MLIRContext *ctx) {
   patterns.insert<ONNXSoftmaxOpLowering>(typeConverter, ctx);
 }
+
+} // namespace onnx_mlir
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/Math/Softmax.cpp:292]: Please fix.

File: src/Conversion/ONNXToKrnl/Math/TopK.cpp [hunk 5]
```diff
@@ -94,3 +96,5 @@ void populateLoweringONNXTopKOpPattern(RewritePatternSet &patterns,
     TypeConverter &typeConverter, MLIRContext *ctx) {
   patterns.insert<ONNXTopKOpLowering>(typeConverter, ctx);
 }
+
+} // namespace onnx_mlir
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/Math/TopK.cpp:100]: Please fix.

File: src/Conversion/ONNXToKrnl/RNN/RNN.cpp [hunk 6]
```diff
@@ -377,3 +379,5 @@ void populateLoweringONNXRNNOpPattern(RewritePatternSet &patterns,
   patterns.insert<ONNXRNNOpLowering<ONNXRNNOp, RnnState, RnnActivationPack,
       RnnWeightPack, RnnBiasPack>>(typeConverter, ctx);
 }
+
+} // namespace onnx_mlir
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/RNN/RNN.cpp:383]: Please fix.

File: src/Conversion/ONNXToKrnl/RNN/RNNBase.cpp [hunk 7]
```diff
@@ -326,3 +328,5 @@ Value emitXSliceAt(ConversionPatternRewriter &rewriter, Location loc, Value X,
 
   return sliceX;
 }
+
+} // namespace onnx_mlir
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/RNN/RNNBase.cpp:332]: Please fix.

File: src/Conversion/ONNXToKrnl/Sequence/SequenceLength.cpp [hunk 8]
```diff
@@ -41,3 +43,5 @@ void populateLoweringONNXSequenceLengthOpPattern(RewritePatternSet &patterns,
     TypeConverter &typeConverter, MLIRContext *ctx) {
   patterns.insert<ONNXSequenceLengthOpLowering>(typeConverter, ctx);
 }
+
+} // namespace onnx_mlir
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/Sequence/SequenceLength.cpp:47]: Please fix.

File: src/Conversion/ONNXToKrnl/ConvertONNXToKrnl.cpp (hunk 2 above)
Comment (etiotto) [reply src/Conversion/ONNXToKrnl/ConvertONNXToKrnl.cpp]: done

File: src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp [hunk 9]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (whitneywhtsang) [inline src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp:1]: remove extra new line.

File: src/Dialect/ONNX/ShapeInference/Transpose.cpp [hunk 10]
```diff
@@ -55,3 +59,5 @@ LogicalResult ONNXTransposeOpShapeHelper::computeShape(
   dimsForOutput(0) = transposedDims;
   return success();
 }
+
+} // namespace onnx_mlir
```
Comment (whitneywhtsang) [inline src/Dialect/ONNX/ShapeInference/Transpose.cpp:63]: Please fix this.

**Review by whitneywhtsang — APPROVED** [review]

File: src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp (hunk 9 above)
Comment (etiotto) [reply src/Conversion/ONNXToKrnl/ControlFlow/Loop.cpp]: done

File: src/Dialect/ONNX/ShapeInference/Transpose.cpp (hunk 10 above)
Comment (etiotto) [reply src/Dialect/ONNX/ShapeInference/Transpose.cpp]: done

**Pull request merged**